subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt

```diff
@@ -40,6 +40,7 @@ class SourcePathProviderTest : FolderBasedTest() {
         assertThat(
             sourcePathFor(
                 classPath = ClassPath.EMPTY,
+                scriptFile = null,
```

Comment:
I think there should be a test in this class covering the recent changes.

Reply:
Fair enough, added some extra tests focusing on the attribute.